pkg/apis/app/v1alpha1/app_deployment_types.go

```diff
@@ -0,0 +1,114 @@
+package v1alpha1
+
+import (
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	kindAppDeployment = "AppDeployment"
+)
+
+// NewAppDeploymentCRD returns a new custom resource definition for AppDeployment.
+// This might look something like the following.
+//
+//     apiVersion: apiextensions.k8s.io/v1beta1
+//     kind: CustomResourceDefinition
+//     metadata:
+//       name: appdeployment.app.giantswarm.io
+//     spec:
+//       group: app.giantswarm.io
+//       scope: Namespaced
+//       version: v1alpha1
+//       names:
+//         kind: AppDeployment
+//         plural: appdeployments
+//         singular: appdeployment
+//
+func NewAppDeploymentCRD() *apiextensionsv1beta1.CustomResourceDefinition {
+	return &apiextensionsv1beta1.CustomResourceDefinition{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiextensionsv1beta1.SchemeGroupVersion.String(),
+			Kind:       "CustomResourceDefinition",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "appdeployments.app.giantswarm.io",
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   "app.giantswarm.io",
+			Scope:   "Namespaced",
+			Version: "v1alpha1",
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Kind:     "AppDeployment",
+				Plural:   "appdeployments",
+				Singular: "appdeployment",
+			},
+			Subresources: &apiextensionsv1beta1.CustomResourceSubresources{
+				Status: &apiextensionsv1beta1.CustomResourceSubresourceStatus{},
+			},
+		},
+	}
+}
+
+func NewAppDeploymentTypeMeta() metav1.TypeMeta {
+	return metav1.TypeMeta{
+		APIVersion: version,
+		Kind:       kindAppDeployment,
+	}
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Example of AppDeployment CR as below.
```

Comment:
To make it a valid godoc:

```suggestion
// AppDeployment CR example as below.
```